pointer-masking-proposal.adoc

```diff
@@ -295,76 +298,133 @@ MASK
+ +
+
 [%header, cols=1*]
 ,===
 spmbase[31:0]
 BASE
 ,===
-Figure 7a: Memory Tagging Extension register (_**spmbase**_) for RV32 (S-MODE) +
+Figure 7c: Memory Tagging Extension register (_**spmbase**_) for RV32 (S-MODE) +
  +
 
 [%header, cols=1*]
 ,===
 upmbase[31:0]
 BASE
 ,===
-Figure 7b: Memory Tagging Extension register (_**upmbase**_) for RV32 (U-MODE) +
+Figure 7d: Memory Tagging Extension register (_**upmbase**_) for RV32 (U-MODE) +
+ +
+
+[%header, cols=1*]
+,===
+mpmbase[63:0]
+BASE
+,===
+Figure 8a: Memory Tagging Extension register (_**mpmbase**_) for RV64 (M-MODE) +
+ +
+
+[%header, cols=1*]
+,===
+hpmbase[63:0]
+BASE
+,===
+Figure 8b: Memory Tagging Extension register (_**hpmbase**_) for RV64 (H-MODE) +
  +
 
 [%header, cols=1*]
 ,===
 spmbase[63:0]
 BASE
 ,===
-Figure 8a: Memory Tagging Extension register (_**spmbase**_) for RV64 (S-MODE) +
+Figure 8c: Memory Tagging Extension register (_**spmbase**_) for RV64 (S-MODE) +
  +
 
 [%header, cols=1*]
 ,===
 upmbase[63:0]
 BASE
 ,===
-Figure 8b: Memory Tagging Extension register (_**upmbase**_) for RV64 (U-MODE) +
+Figure 8d: Memory Tagging Extension register (_**upmbase**_) for RV64 (U-MODE) +
+ +
+
+[%header, cols=1*]
+,===
+mpmbase[127:0]
+BASE
+,===
+Figure 9a: Memory Tagging Extension register (_**mpmbase**_) for RV128 (M-MODE) +
+ +
+
+[%header, cols=1*]
+,===
+hpmbase[127:0]
+BASE
+,===
+Figure 9b: Memory Tagging Extension register (_**hpmbase**_) for RV128 (H-MODE) +
  +
 
 [%header, cols=1*]
 ,===
 spmbase[127:0]
 BASE
 ,===
-Figure 9a: Memory Tagging Extension register (_**spmbase**_) for RV128 (S-MODE) +
+Figure 9c: Memory Tagging Extension register (_**spmbase**_) for RV128 (S-MODE) +
  +
 
 [%header, cols=1*]
 ,===
 upmbase[127:0]
 BASE
 ,===
-Figure 9b: Memory Tagging Extension register (_**upmbase**_) for RV128 (U-MODE) +
+Figure 9d: Memory Tagging Extension register (_**upmbase**_) for RV128 (U-MODE) +
  +
 
 [[explanation]]
 ## Explanation
 
-_**MMTE**_ register fully two-fold function:
+_**MMTE**_ register fulfills two-fold function:
 
-1. Can be programmable by more privileged mode
+1. Can be programmable by more privileged mode (unless PM.Current bit is enabled)
 2. Performs status register function for the current privilege mode
 
-_PM_ bits from **_MMTE_** register are accessible in all modes (U/S/HS/M) and can be read to query the current status of PM feature (if enabled). However, only M-mode privileged code can set the value for _PM_ bits.
+_PM_ bits from **_MMTE_** register are accessible in all modes (U/S/HS/M) and can be read to query the current status of PM feature (if enabled). By default, only higher privileged code can set the value for _PM_ bits. However, higher privileged code can enable _PM.Current_ bit for lower privileged code. In such scenario, current privilege code has a possibility to self-manage its own configuration of _PM_ bits.
+If _PM.Enable_ bit is changing the status, corresponding _xTE_ bit is automatically updated.
```

Comment:
From the text it's unclear, what comes first: enabling PM.Enable or xTE? Could you possibly provide several examples of the flow with these registers and expected behavior?

Reply:
Be aware, this is still a draft of the proposal and we are still polishing various corner cases.

The purpose of xTE bit is to choose which pair of mask/base to use. However, PM.enable bit from the corresponding level defines if they are enforced. During CPU-mode switch appropriate xTE bit is set (and other bits are disabled). However, user can choose to manually enable that xTE bit which he wants. Enforcement of base/mask is done only when PM.enable is set to 1